tests/ga/test_update.py

```diff
@@ -1701,34 +1692,36 @@ def setUp(self):
         AgentTestCase.setUp(self)
         self.event_patch = patch('azurelinuxagent.common.event.add_event')
         currentThread().setName("ExtHandler")
+        protocol = Mock()
+        protocol.get_ext_handlers = Mock(return_value=(Mock(), Mock()))
         self.update_handler = get_update_handler()
         self.update_handler.protocol_util = Mock()
+        self.update_handler.protocol_util.get_protocol = Mock(return_value=protocol)
         clear_singleton_instances(ProtocolUtil)
 
     def _test_run(self, invocations=1):
-        iterations = [0]
-
-        def iterator(*args, **kwargs):  # pylint: disable=unused-argument
-            iterations[0] += 1
-            if iterations[0] >= invocations:
-                self.update_handler.running = False
-            return
+        def iterator(*_, **__):
+            iterator.count += 1
+            if iterator.count <= invocations:
+                return True
+            return False
+        iterator.count = 0
 
         with patch('os.getpid', return_value=42):
             with patch.object(UpdateHandler, '_is_orphaned') as mock_is_orphaned:
                 mock_is_orphaned.__get__ = Mock(return_value=False)
-                with patch('azurelinuxagent.ga.exthandlers.get_exthandlers_handler'):
```

Comment:
those tests were mocking the global sleep(), which make them extremely fragile. They broke. Now they mock is_running.

BTW - Note the amount of mocks in these tests. THIS IS NOT THE WAY TO WRITE TESTS.